components/director/internal/domain/runtime/selfreg_manager.go

```diff
@@ -134,9 +143,9 @@ func (s *selfRegisterManager) GetSelfRegDistinguishingLabelKey() string {
 	return s.cfg.SelfRegisterDistinguishLabelKey
 }
 
-func (s *selfRegisterManager) createSelfRegPrepRequest(runtimeID, tenant string) (*http.Request, error) {
+func (s *selfRegisterManager) createSelfRegPrepRequest(runtimeID, tenant string, instanceConfig InstanceConfig) (*http.Request, error) {
```

Comment:
we can pass only URL string instead of whole config?
P.S. Valid for `createSelfRegDelRequestv` as well